### PR TITLE
MAKE-997: add time-based iterators

### PR DESCRIPTION
**Pull request by dbradf**
I noticed we are doing this quite a bit. So it seems useful to put in this library.

### CODE REVIEW

File: src/evergreen/__init__.py [hunk 1]
```diff
@@ -1,5 +1,21 @@
 """Evergreen API Module."""
 from pylibversion import version_tuple_to_str
 
-VERSION = (0, 6, 12)
+VERSION = (0, 6, 13)
 __version__ = version_tuple_to_str(VERSION)
+
+
+# Shortcuts for importing.
```
Comment (dbradf) [inline src/evergreen/__init__.py]: This seems like common practice in python. It should make importing the project a lot easier.

Comment (dbradf) [reply src/evergreen/__init__.py]: Yes, it allows importing everything from the evergreen module:
```
from evergreen import EvergreenApi, Version, Patch
```

File: src/evergreen/util.py [hunk 2]
```diff
@@ -63,3 +63,30 @@ def parse_evergreen_date(evg_date):
...
+    Iterate over a window of time.
+
+    For a given iterator, generate the items that are within the specified time window.
+
+    Note: Since most evergreen iterators start with the most recent items and then look backwards
+    in time, `start` and `end` refer to the start and end of when items will be seen (i.e. `start`
+    should be later in time than `end` since we will start seeing new items first.
+
+    :param iterator: Iterator to window.
+    :param start: Timestamp to start returning items (later in time than end).
```
[the remark this answers is not in the document]
Comment (dbradf) [reply src/evergreen/util.py]: I like that a lot more. Changed.

File: tests/evergreen/test_api.py [hunk 3]
```diff
@@ -162,13 +165,67 @@ def test_version_by_project(self, mocked_api):
...
+            deepcopy(sample_version),
+            deepcopy(sample_version),
+            deepcopy(sample_version),
+        ]
+        # Create a window of 1 day, and set the dates so that only the middle items should be
+        # returned.
+        one_day = timedelta(days=1)
+        one_hour = timedelta(hours=1)
+        start_date = parse_evergreen_datetime(version_list[1]['create_time'])
+        end_date = start_date - one_day
```
Comment (lydiastepanek) [inline tests/evergreen/test_api.py]: Could we also rename `start_date` -> `before_date` and `end_date` -> `after_date`?

Comment (dbradf) [reply tests/evergreen/test_api.py]: yes, updated.

**Comment by dbradf** [comment]
evergreen merge

**Pull request merged**